internal/engine/executor.go

```diff
@@ -198,7 +199,14 @@ func (e *Executor) HandleEntityEvent(msg *message.Message) error {
 	return nil
 }
 func (e *Executor) prepAndEvalEntityEvent(ctx context.Context, inf *entities.EntityInfoWrapper) error {
-	provider, err := e.providerStore.GetByName(ctx, inf.ProjectID, inf.Provider)
+	// TODO: clean this up once we get rid of provider name
+	var provider *db.Provider
+	var err error
+	if inf.ProviderID == uuid.Nil {
+		provider, err = e.providerStore.GetByName(ctx, inf.ProjectID, inf.Provider)
+	} else {
+		provider, err = e.providerStore.GetByID(ctx, inf.ProviderID)
+	}
```

Comment:
Slight preference to reverse the order of these conditions, since we want to prefer the `inf.ProviderID` not being nil.

Reply:
I will do this in the next PR where the ProviderID is actually set. Right now, since it's not being set, the current order makes more sense.